langserver/references.go

```diff
@@ -105,6 +105,11 @@ func (h *LangHandler) handleTextDocumentReferences(ctx context.Context, conn jso
 		close(locsC)
 	}()
 
+	// Don't include decl if it is outside of workspace.
+	if params.Context.IncludeDeclaration && PathHasPrefix(defpkg, h.init.RootImportPath) {
```

Comment:
I think this needs to happen before (or in) the goroutine. Otherwise it is racing with the close on the channel.

Since the declaration requires no computation, it seems like it should be first anyway?